src/lib.rs

```diff
@@ -316,6 +316,30 @@ impl<T> Sender<T> {
         }
     }
 
+
+    /// Returns true if the associated Receiver handle has been dropped.
+    ///
+    /// If true is returned, a future call to send is guaranteed to return an error.
+    pub fn is_closed(&self) -> bool {
+        // SAFETY: Since we are holding a Sender's reference, we know that:
+        // - `send(..)` has not been called yet
+        // - `Sender` has not been dropped.
+        //
+        // This is sufficient to guarantee that the channel is still on the heap.
+        //
+        // Note that if the receiver disconnects it does not free the channel.
+        let channel = unsafe { self.channel_ptr.as_ref() };
+
+        // ORDERING: We *chose* a Relaxed ordering here as it sufficient to
+        // enforce the method's contract: "if true is returned, a future
+        // call to send is guaranteed to return an error."
+        //
+        // It is possible however to craft a theoretical scenario where
+        // the Relaxed ordering would have visible effects for the client
+        // of this library.
```

Comment:
This is very vague. What are these theoretical scenarios? What effects? I think we can discuss those in this PR thread and maybe leave it out of the source code, unless it's important for the reader to know about.

Since we give no guarantees about "if this method returns false, the sender is still alive" I think relaxed is probably good enough. But if you have a scenario where a stronger ordering would be beneficial we can of course consider that!

Reply:
I was thinking:

I don't have in mind a scenario where it is beneficial.
I just have a scenario where it makes a difference.

```
thread 1: drop(rx1); drop(rx2)
thread 2: let rx2_closed = rx2.is_closed(); let rx1_closed = tx1.is_closed();
```

With Relaxed I believe it is possible to observe
`(rx1_closed, rx2_closed)` taking the value `(false, true)`.
while it is not possible to observe that with `Acquire`.